src/mainwindow.cpp

```diff
@@ -1074,3 +1076,42 @@ void MainWindow::RefilterTreeView()
     model->layoutChanged();
     view->scrollTo(previousIdx, QAbstractItemView::PositionAtCenter);
 }
+
+bool MainWindow::eventFilter(QObject* obj, QEvent* event)
+{
+    if (obj == tabWidget->tabBar() &&
+        event->type() == QEvent::MouseButtonPress)
+    {
+        auto mouseEvent = static_cast<QMouseEvent *>(event);
+        int idx = tabWidget->tabBar()->tabAt(mouseEvent->pos());
+
+        if (mouseEvent->button() == Qt::MidButton)
+        {
+            on_tabWidget_tabCloseRequested(idx);
+            return true;
+        }
+        else if (mouseEvent->button() == Qt::RightButton)
+        {
+            TreeModel* model = GetTreeModel(GetTreeView(idx));
+            if (!model || model->TabType() == TABTYPE::ExportedEvents)
+                return true;
+
+            LogTab* logTab = m_logTabs[model];
+
+            QAction* actionCopyFullPath = new QAction("Copy full path", this);
```

Comment:
Could this leak? I don't think it is cleaned until the main window is closed, which is too late.